test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/DrainModeResumeEndToEndTests.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.Azure.WebJobs.Script.WebHost.Models;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.WebJobs.Script.Tests;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.EndToEnd
+{
+    [Trait(TestTraits.Category, TestTraits.EndToEnd)]
+    [Trait(TestTraits.Group, TestTraits.DrainModeEndToEnd)]
+    public class DrainModeResumeEndToEndTests : ResumeTestFixture
+    {
+        [Fact]
+        public async Task DrainModeEnabled_RunningHost_StartsNewHost_ReturnsOk()
+        {
+            // Validate the drain state is "Disabled" initially
+            var response = await SamplesTestHelpers.InvokeDrainStatus(this);
+            var responseString = response.Content.ReadAsStringAsync().Result;
+            var status = JsonConvert.DeserializeObject<DrainModeStatus>(responseString);
+
+            Assert.Equal(status.State, DrainModeState.Disabled);
+
+            ManualResetEvent resetEvent = new ManualResetEvent(false);
+            _ = Task.Run(async () =>
```

Comment:
Why do you need to use Task.Run here? I think it'd be cleaner if you removed it. The operations will run in parallel with your http trigger request below regardless.

Reply:
Hmm something funky was happening without this (it's been a minute so I don't remember the detials), I'll look into doing it without Task.Run again and see if it works/ what the issue is

Reply:
Wasn't able to repro the issue I was having last time so I'm removing Task.Run as suggested, thank you!